streams/src/test/java/org/apache/kafka/streams/processor/internals/PartitionGroupTest.java

```diff
@@ -266,4 +293,20 @@ private void verifyBuffered(final int totalBuffered, final int partitionOneBuffe
         assertEquals(partitionOneBuffered, group.numBuffered(partition1));
         assertEquals(partitionTwoBuffered, group.numBuffered(partition2));
     }
+
+    @Test
+    public void testSetPartitionTimestamp() {
+        group.setPartitionTime(partition1, 100L);
+        assertEquals(100L, group.partitionTimestamp(partition1));
+        assertThrows(errMessage, NullPointerException.class, () -> {
+            group.setPartitionTime(randomPartition, 0L);
+        });
+    }
```

Comment:
This tests misses to verify whether `streamTime` is set or not.

Furthermore, I would write two (or three) distinct tests:
- `partitionTimestamp` is set (could be further split for `streamTime` is set or not)
- `NullPointerException` is thrown